src/shell/focus/mod.rs

```diff
@@ -231,29 +231,29 @@ fn update_focus_state(
             if target.is_some() {
                 //need to borrow mutably for surface under
                 let mut shell = state.common.shell.write().unwrap();
-                // get the top left corner of the target element
+                // get geometry of the target element
                 let geometry = shell.focused_geometry(target.unwrap());
-                //to avoid the nested mutable borrow of state
-                if geometry.is_some() {
-                    let top_left = geometry.unwrap().loc.to_f64();
+                if let Some(geometry) = geometry {
+                    // get the center of the target element
+                    let new_pos = (geometry.loc + geometry.size.downscale(2).to_point()).to_f64();
```

Comment:
This is quite the nitpick, but `downscale(2).to_point()` doesn't really communicate the intent of this operation at all. Please do a more obvious devision by 2 or add a comment.

Reply:
done, i think